frontend/src/main/components/Jobs/MilkCowsJobForm.js

```diff
@@ -1,22 +1,62 @@
 import { Button, Form } from "react-bootstrap";
+import { useState } from "react";
 import { useForm } from "react-hook-form";
+import { useBackend } from "main/utils/useBackend";
+import CommonsSelect from "main/components/Commons/CommonsSelect";
 
-function MilkTheCowsForm( {submitAction} ) {
-
-    // Stryker disable all
-    const {
-      handleSubmit,
-    } = useForm(
-    );
-    // Stryker restore all
-
-    return (
-      <Form onSubmit={handleSubmit(submitAction)}>
-        <p>Click this button to milk the cows across all commons!</p>
-        <Button type="submit" data-testid="MilkTheCowsForm-Submit-Button">Milk!</Button>
-    </Form>
-    );
+function MilkTheCowsForm( { submitAction, testid = "MilkTheCowsForm" } ) {
+
+  // Stryker restore all
+
+  const { data: commonsAll } = useBackend(
+    ["/api/commons/all"],
+    { url: "/api/commons/all" },
+    []
+  );
+
+  const allCommonsProp = {"id":0,"name":"All Commons"}
+  
+  const commons = [allCommonsProp, ...commonsAll]
+
+  const [selectedCommons, setSelectedCommons] = useState(null);
+  const [selectedCommonsName, setSelectedCommonsName] = useState(null);
+
+  const {
+    handleSubmit,
+  } = useForm();
+
+  const handleCommonsSelection = (id, name) => {
+    setSelectedCommons(id);
+    setSelectedCommonsName(name);
+  };
+
+  const onSubmit = () => {
+    const params = { selectedCommons, selectedCommonsName };
+    submitAction(params);
+  };
+
+  if (selectedCommons === null) {
+    setSelectedCommons(commons[0].id);
+    setSelectedCommonsName(commons[0].name);
   }
```

Comment:
Avoid setting state in render unless under *very* special circumstances. Set initial values in useState or fallback to default values.